pages/myProfile/index.js

```diff
@@ -14,46 +16,110 @@ const myProfile = ({loginName, userArray}) => {
     },
   };
   // parses loginName String to Object to be able to compare them in the checkLoggedInUser Method
-  const loginNameObject = {name: `${loginName}`};
-  const checkLoggedInUser = userArray.filter(user => {
-    return user.name == loginNameObject.name;
+  const [ownProfileUser, setOwnProfileUser] = useState();
+  useEffect(() => {
+    //first
+    const loginNameObject = {name: `${loginName}`};
+    const checkLoggedInUser = userArray.filter(function (user) {
+      return user.name == loginNameObject.name;
+    });
+    let singleObject = {};
+    try {
+      singleObject = {
+        id: checkLoggedInUser[0].id,
+        name: checkLoggedInUser[0].name,
+        address: checkLoggedInUser[0].address,
+        email: checkLoggedInUser[0].email,
+        phone: checkLoggedInUser[0].phone,
+        website: checkLoggedInUser[0].website,
+        image: checkLoggedInUser[0].image,
```

Comment:
If you would use find method above you don't need an index at this point, because you already got the right object:

```suggestion
        id: checkLoggedInUser.id,
        name: checkLoggedInUser.name,
        address: checkLoggedInUser.address,
        email: checkLoggedInUser.email,
        phone: checkLoggedInUser.phone,
        website: checkLoggedInUser.website,
        image: checkLoggedInUser.image,
```

Reply:
Fixed and pushed new version.